core/src/main/java/org/apache/carbondata/core/util/CarbonMetadataUtil.java

```diff
@@ -368,7 +368,13 @@ private static int compareMeasureData(byte[] first, byte[] second, DataType data
       secondBuffer.put(second);
       firstBuffer.flip();
       secondBuffer.flip();
-      return (int) (firstBuffer.getDouble() - secondBuffer.getDouble());
+      long compare = firstBuffer.getLong() - secondBuffer.getLong();
```

Comment:
For double type we should call getDouble

Reply:
Yes , its a copy paste mistake :)

Reply:
:)